static/js/publisher-pages/pages/Builds/RepoConnected.tsx

```diff
@@ -0,0 +1,314 @@
+      },
+    });
+
+    if (!response.ok) {
+      throw new Error("Unable to trigger new build");
+    }
+
+    refetch();
+    setAutoTriggerBuild(false);
+  }
+
+  if (autoTriggerBuild) {
+    trigger();
+  }
+
+  useEffect(() => {
+    setTimeout(() => {
+      refetch();
+    }, 30000);
+  }, []);
+
+  return (
+    <>
+      <Strip shallow className="u-no-padding--top u-no-padding--bottom">
+        <div className="snapcraft-p-sticky js-sticky-bar">
+          <div className="u-fixed-width">
+            {githubData !== null && (
+              <p>
+                Your snap is linked to:{" "}
+                <a href={`https://github.com/${githubData.github_repository}`}>
+                  {githubData.github_repository}
+                </a>{" "}
+                |{" "}
+                <Button
+                  appearance="link"
+                  onClick={() => {
+                    setDisconnectModalOpen(true);
+                  }}
+                >
+                  Disconnect repo
+                </Button>
+              </p>
+            )}
+          </div>
+        </div>
+      </Strip>
+      <Row>
+        <Col size={6}>
+          <h2 className="p-heading--4">Latest builds</h2>
+        </Col>
+        <Col size={6} className="u-align--right">
+          <Button
+            disabled={triggeringBuild}
+            onClick={async () => {
+              setTriggeringBuild(true);
+              const response = await fetch(
+                `/api/${snapId}/builds/trigger-build`,
+                {
+                  method: "POST",
+                  headers: {
+                    "Content-Type": "application/json",
+                    "X-CSRFToken": window.CSRF_TOKEN,
+                  },
+                },
+              );
+
+              if (!response.ok) {
+                throw new Error("Unable to trigger new build");
+              }
+
+              refetch();
+            }}
+          >
+            {triggeringBuild ? (
+              <>
+                <i className="p-icon--spinner u-animation--spin"></i>
+                &nbsp;Requesting
+              </>
+            ) : (
+              "Trigger new build"
+            )}
+          </Button>
+        </Col>
+      </Row>
+      {isLoading && <>Waiting for builds...</>}
+      {isFetched && data && (
+        <>
+          <MainTable
+            sortable
+            headers={[
+              { content: "ID", sortKey: "id" },
+              { content: "Architecture", sortKey: "arch" },
+              { content: "Build duration", sortKey: "duration" },
+              {
+                content: "Result",
+                sortKey: "result",
+                className: "p-table__cell--icon-placeholder",
+              },
+              {
+                content: "Build finished",
+                sortKey: "status",
+                className: "u-align--right",
+              },
+            ]}
+            rows={data.snap_builds.map(
```

Comment:
should we assign it to a variable to increase the readability?

Reply:
`data` is only guaranteed to be defined within this scope so we'd need to add some logic here to handle that if we used a variable, which I don't see adds any benefits?